fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java

```diff
@@ -88,10 +88,8 @@ public void generateClientSchedule() {
                                 next = start.plusDays((int) (long) adhoc.getReportRunEvery());
                                 run = Days.daysBetween(start, end).getDays() >= adhoc.getReportRunEvery();
                                 break;
-                            default:
-                                throw new IllegalStateException();
                         }
-
+                        throw new IllegalStateException();
```

Comment:
wait... wat ??  By doing this, it will ALWAYS throw, no? This will completely break. @awasum please help..

Reply:
@percyashu ..Maybe remove this exception? and put it back in the default case inside the } on line 91 above?

Also if you want to add a default step, you need to check the `ReportRunFrequency.fromId(adhoc.getReportRunFrequency()` method from line 70 above and see how many different cases needs to be treated, then try to handle them all. Do we absolutely need a default case in all implementations of Switch statement?
As seen here: https://errorprone.info/bugpattern/UnnecessaryDefaultInEnumSwitch, dont put a default statement if all the cases have been handled. Maybe skip it? Will that throw an error?

Or given that ReportRunFrequency is an enum and we/Fineract is already handling all the 5 cases here, we can just throw a `new AdHocScheduledJobFrequencyException()` error here as default case with a message like "Correct Frequency not specified, Check the API docs"... @vorburger  @ptuomola your take?

Reply:
Isn't this ErrorProne check specifically highlighting switch statements where the default case is never used, because we are already handling all the cases specifically? Given that, wouldn't the right solution just be to remove the default including the throw, and not add anything?

Reply:
@ptuomola I thought so to but it requires a return. 
@vorburger I thought if the case is not found, switch default is used with exception thrown and if case is not found and no default then switch is not executed then exception is thrown. That's why I did it like this. Just undid what I did and added a suppressedwarning